crates/external_services/src/hashicorp_vault/core.rs

```diff
@@ -0,0 +1,227 @@
+//! Interactions with the HashiCorp Vault
+
+use std::{collections::HashMap, future::Future, pin::Pin};
+
+use common_utils::{ext_traits::ConfigExt, fp_utils::when};
+use error_stack::{Report, ResultExt};
+use masking::Secret;
+use vaultrs::client::{VaultClient, VaultClientSettingsBuilder};
+
+static HC_CLIENT: tokio::sync::OnceCell<HashiCorpVault> = tokio::sync::OnceCell::const_new();
```

Comment:
We can use `once_cell::sync::OnceCell` here, and make `get_hashicorp_client()` a sync function.

Reply:
Same here, will take this up in next PR